### PR TITLE
Update server tutorial codebase, fixes #333

### LINKED ISSUE #333: Server tutorial has outdated syntax
https://swiftpackageindex.com/apple/swift-openapi-generator/0.3.0/tutorials/swift-openapi-generator/serverswiftpm

`.text` doesn't build, should be `.plainText`. We should search the repo for other occurrences.

And the string needs to be wrapped in HTTPBody.

### PR DESCRIPTION
**Pull request by tib**
This pull request aims to fix the server tutorial sample code.
Sample snippets updated to use Swift 5.9 and Swift OpenAPI generator tools 0.3.0+

### CODE REVIEW

**Comment by czechboy0** [comment]
This is great, @tib, thank you!

Would you be open to also updating the other 2 tutorials, to keep all 3 in sync?

**Comment by czechboy0** [comment]
@swift-server-bot add to allowlist

File: Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift [hunk 1]
```diff
@@ -17,7 +17,9 @@ struct GreetingServiceAPIImpl: APIProtocol {
         _ input: Operations.getEmoji.Input
     ) async throws -> Operations.getEmoji.Output {
         let emojis = "👋👍👏🙏🤙🤘"
-        return .ok(.init(body: .text(String(emojis.randomElement()!))))
+        let emoji = String(emojis.randomElement()!)
+        let data = emoji.data(using: .utf8)!
+        return .ok(.init(body: .plainText(.init(data))))
```
Comment (czechboy0) [inline Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: You can provide the string directly to the HTTPBody initializer, no need to explicitly concert to data first.

Comment (tib) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: I did that, but the response was a `?` character in that case, must be an utf8 issue... 🤔

**Comment by czechboy0** [comment]
> > This is great, @tib, thank you!
> > Would you be open to also updating the other 2 tutorials, to keep all 3 in sync?
> 
> Yes, of course I can do that, do you want me to open separate PRs or alter this one?

You can do it in this one.

File: Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift (hunk 1 above)
Comment (czechboy0) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: Can you still reproduce it? I'd expect the code to work if it's:

```swift
let emojis = "👋👍👏🙏🤙🤘"
let emoji = String(emojis.randomElement()!)
return .ok(.init(body: .plainText(.init(emoji))))
```

If it doesn't, we should investigate.

Comment (tib) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: This is what I've got when I don't convert to utf8 data:

```
tib@~: curl -i http://localhost:8080/api/emoji
HTTP/1.1 200 OK
Content-Type: text/plain
content-length: 1
connection: keep-alive
date: Mon, 23 Oct 2023 12:21:53 GMT

?%                                                                                                                                      tib@~: 
```

Comment (tib) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: It works with Hummingbird, but gives the ? when using Vapor, so it's going to be an issue in the transport layer... 😅

Comment (tib) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: This change would fix the runtime. I already made a fork, can I open a PR?

Comment (czechboy0) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: Yes, that looks like a bug. Please file a separate issue and let's fix that first, please? Great catch. Can you open a PR to the runtime?

Comment (czechboy0) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: I think you can just remove the `length` parameter, and the overload that takes a collection of bytes will compute the length correctly.

Comment (tib) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: Yeah, that worked opened a [PR](https://github.com/apple/swift-openapi-runtime/pull/60), also conformed the Length to Equatable so we can have nice tests, not sure if you agree with that or not, but feel free to discuss this there. 🙃

Comment (czechboy0) [reply Sources/swift-openapi-generator/Documentation.docc/Tutorials/_Resources/server.main.2.swift:22]: It's reasonable and I see how it makes the tests nicer, so let's keep it.

**Comment by czechboy0** [comment]
Ok with https://github.com/apple/swift-openapi-runtime/releases/tag/0.3.4 released, please see if you can simplify the example and it actually works for you :)

**Comment by tib** [comment]
> Ok with https://github.com/apple/swift-openapi-runtime/releases/tag/0.3.4 released, please see if you can simplify the example and it actually works for you :)

```swift
let emoji = String(emojis.randomElement()!)
return .ok(.init(body: .plainText(.init(emoji))))
```

I already changed that. 😉

**Review by czechboy0 — APPROVED** [review]

**Pull request merged**